### PR TITLE
driver: counter: m41t62 Added support

### PR DESCRIPTION
**Pull request by ivonalo**
Added driver support for the M41T62 RTC
Signed-off-by: Ivona Loncar <ivona.loncar@greyp.com>

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thank you for the contribution!

Why did you implement this as a sensor driver? I think it's better suited as a counter driver instead.

**Comment by ivonalo** [comment]
> Thank you for the contribution!
> 
> Why did you implement this as a sensor driver? I think it's better suited as a counter driver instead.

Thank you for your answer! I have refactored this pull request into a counter driver, so please take a look

File: drivers/sensor/m41t62/m41t62.c [hunk 1]
```diff
@@ -0,0 +1,747 @@
...
+	alarm_buf[0] &= ~(M41T62_ALMONTH10_MASK | M41T62_ALMONTH_MASK);
+	alarm_buf[0] |= write_buf[0];
+
+	/* If the last address written is the Alarm Seconds, the address
+	 * pointer will increment to the address of the Flags register.
+	 * Therefore an alarm condition will not cause the interrupt/flag to occur.
+	 * That is why the write addresses are decreasing in this case and therefore
+	 * the i2c_burst_write function can't be used.
+	 */
+	for (int i = 0; i < ARRAY_SIZE(alarm_buf); ++i) {
```
Comment (gmarull) [inline drivers/sensor/m41t62/m41t62.c]: `size_t i`

File: drivers/sensor/m41t62/m41t62.h [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+};
+
+/* @brief  M41T62 Alarm repeat mode */
+enum {
+	M41T62_OnceperSecond    = (uint8_t)31,
+	M41T62_OnceperMinute    = (uint8_t)30,
+	M41T62_OnceperHour      = (uint8_t)28,
+	M41T62_OnceperDay       = (uint8_t)24,
+	M41T62_OnceperMonth     = (uint8_t)16,
+	M41T62_OnceperYear      = (uint8_t)0
```
Comment (gmarull) [inline drivers/sensor/m41t62/m41t62.h]: naming should be uppercase, `uint8_t` is also redundant here

File: include/drivers/sensor.h [hunk 3]
```diff
@@ -174,6 +174,16 @@ enum sensor_channel {
...
 
+	SENSOR_CHAN_RTC_CONTROL_BITS,
+
+	SENSOR_CHAN_RTC_TIME,
+
+	SENSOR_CHAN_RTC_ALARM_TIME,
+
+	SENSOR_CHAN_SQUARE_WAVE_FREQ,
+
+	SENSOR_CHAN_RESET_OSCILLATOR,
```
Comment (gmarull) [inline include/drivers/sensor.h]: Channels should be documented, since they don't look "standard" I'm not sure if they should be "private", @MaureenHelm ?

**Review by gmarull — CHANGES_REQUESTED** [review]
I did an initial pass, requested a few changes. I see a sensor driver and then a counter... Can this be clarified?

**Comment by ivonalo** [comment]
> I did an initial pass, requested a few changes. I see a sensor driver and then a counter... Can this be clarified?

@gmarull the sensor driver was a commit left from the previous version. Now I have removed that commit and there is only the counter driver left as discussed with @MaureenHelm. Please take a look

**Comment by ivonalo** [comment]
> > Thank you for the contribution!
> > Why did you implement this as a sensor driver? I think it's better suited as a counter driver instead.
> 
> @MaureenHelm  thank you for your answer! I have refactored this pull request into a counter driver, so please take a look

**Pull request closed without merging**